engine/src/main/java/org/terasology/world/block/tiles/WorldAtlasImpl.java

```diff
@@ -275,6 +303,23 @@ private void createTextureAtlas(final Texture texture) {
         return data;
     }
 
+    // Ref: http://stackoverflow.com/questions/221830/set-bufferedimage-alpha-mask-in-java/8058442#8058442
+    public void storeGreyscaleMapIntoAlpha(BufferedImage imageWithoutAlpha, BufferedImage greyscaleImage) {
+        int width = imageWithoutAlpha.getWidth();
+        int height = imageWithoutAlpha.getHeight();
+
+        int[] imagePixels = imageWithoutAlpha.getRGB(0, 0, width, height, null, 0, width);
```

Comment:
I'm wondering here if it'd be worth to use the BufferedImage.getData() method to obtain a Raster object and then iterate over pixels directly instead of going through integer arrays. Not quite a priority though.